Source/ASCollectionView.mm

```diff
@@ -966,14 +968,8 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
   if (shouldDequeueExternally) {
     view = [(id<ASCollectionDataSourceInterop>)_asyncDataSource collectionView:collectionView viewForSupplementaryElementOfKind:kind atIndexPath:indexPath];
   } else {
-    view = [self dequeueReusableSupplementaryViewOfKind:kind withReuseIdentifier:kReuseIdentifier forIndexPath:indexPath];
-  }
-
-  if (!node.shouldUseUIKitCell) {
     ASDisplayNodeAssert(node != nil, @"Supplementary node should exist.  Kind = %@, indexPath = %@, collectionDataSource = %@", kind, indexPath, self);
-  }
-
-  if (node) {
+    view = [self dequeueReusableSupplementaryViewOfKind:kind withReuseIdentifier:kReuseIdentifier forIndexPath:indexPath];
```

Comment:
Basically the change here is to only call the _rangeController configure in the case that we have a valid node to use, otherwise rely on UICollectionView doing its normal thing.